### PR TITLE
Updated Albert model Card

### PR DESCRIPTION
**Pull request by souvikchand**
# What does this PR do?

Updates `ALBERT` model card as per #[36979](https://github.com/huggingface/transformers/issues/36979)

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)?  https://github.com/huggingface/transformers/issues/36979
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?
@stevhliu Please check the PR and see if it's alright :smile:

### CODE REVIEW

File: docs/source/en/model_doc/albert.md [hunk 1]
```diff
@@ -14,100 +14,144 @@ rendered properly in your Markdown viewer.
...
+print(f"Size after: {quantized_model.get_memory_footprint()/1e6:.1f}MB")
+
+# Usage example
+inputs = tokenizer("Albert Einstein was born in [MASK].", return_tensors="pt")
+with torch.no_grad():
+    outputs = quantized_model(**inputs)
+    print(tokenizer.decode(outputs.logits[0].argmax(-1)))
+```
+
+> ALBERT is not compatible with `AttentionMaskVisualizer` as it uses masked self-attention rather than causal attention. So don't have `_update_causal_mask` method.
```
Comment (stevhliu) [inline docs/source/en/model_doc/albert.md:145]:
```suggestion
```

File: docs/source/en/model_doc/albert.md [hunk 2]
```diff
@@ -14,100 +14,144 @@ rendered properly in your Markdown viewer.
...
 
-This model was contributed by [lysandre](https://huggingface.co/lysandre). This model jax version was contributed by
-[kamalkraj](https://huggingface.co/kamalkraj). The original code can be found [here](https://github.com/google-research/ALBERT).
+## Notes
 
+- All tokens attend to all others (like BERT).
```
Comment (stevhliu) [inline docs/source/en/model_doc/albert.md:149]:
```suggestion
- Inputs should be padded on the right because BERT uses absolute position embeddings.
- The embedding size `E` is different from the hidden size `H` because the embeddings are context independent (one embedding vector represents one token) and the hidden states are context dependent (one hidden state represents a sequence of tokens). The embedding matrix is also larger because `V x E` where `V` is the vocabulary size. As a result, it's more logical if `H >> E`. If `E < H`, the model has less parameters.
```

File: docs/source/en/model_doc/albert.md [hunk 3]
```diff
@@ -14,100 +14,144 @@ rendered properly in your Markdown viewer.
...
+- ALBERT supports a maximum sequence length of 512 tokens.
+- Cannot be used for autoregressive generation (unlike GPT)
+- ALBERT requires absolute positional embeddings, and it expects right-padding (i.e., pad tokens should be added at the end, not the beginning).
+- ALBERT uses token_type_ids, just like BERT. So you should indicate which token belongs to which segment (e.g., sentence A vs. sentence B) when doing tasks like question answering or sentence-pair classification.
+- ALBERT uses a different pretraining objective called Sentence Order Prediction (SOP) instead of Next Sentence Prediction (NSP), so fine-tuned models might behave slightly differently from BERT when modeling inter-sentence relationships.
```
Comment (stevhliu) [inline docs/source/en/model_doc/albert.md:154]:
```suggestion
```

File: docs/source/en/model_doc/albert.md [hunk 4]
```diff
@@ -14,100 +14,144 @@ rendered properly in your Markdown viewer.
...
-  number of (repeating) layers.
-- Embedding size E is different from hidden size H justified because the embeddings are context independent (one embedding vector represents one token), whereas hidden states are context dependent (one hidden state represents a sequence of tokens) so it's more logical to have H >> E. Also, the embedding matrix is large since it's V x E (V being the vocab size). If E < H, it has less parameters.
-- Layers are split in groups that share parameters (to save memory).
-Next sentence prediction is replaced by a sentence ordering prediction: in the inputs, we have two sentences A and B (that are consecutive) and we either feed A followed by B or B followed by A. The model must predict if they have been swapped or not.
+<hfoptions id="usage">
+<hfoption id="Pipeline>
 
-### Using Scaled Dot Product Attention (SDPA)
+```py 
+import torch
```
Comment (stevhliu) [inline docs/source/en/model_doc/albert.md:50]: You can simplify the whole thing to:

```py
import torch
from transformers import pipeline

pipeline = pipeline(
    task="fill-mask",
    model="albert-base-v2",
    torch_dtype=torch.float16,
    device=0
)
pipeline("Plants create [MASK] through a process known as photosynthesis.", top_k=5)
```

File: docs/source/en/model_doc/albert.md [hunk 5]
```diff
@@ -14,100 +14,144 @@ rendered properly in your Markdown viewer.
...
 ```
-from transformers import AlbertModel
-model = AlbertModel.from_pretrained("albert/albert-base-v1", torch_dtype=torch.float16, attn_implementation="sdpa")
-...
+
+</hfoption>
+<hfoption id="AutoModel">
+
+```py
+import torch
```
Comment (stevhliu) [inline docs/source/en/model_doc/albert.md:72]: This can also be simplified a bit without the comments and include SDPA:

```py
import torch
from transformers import AutoModelForMaskedLM, AutoTokenizer

tokenizer = AutoTokenizer.from_pretrained("albert/albert-base-v2")
model = AutoModelForMaskedLM.from_pretrained(
    "albert/albert-base-v2",
    torch_dtype=torch.float16,
    attn_implementation="sdpa",
    device_map="auto"
)

prompt = "Plants create energy through a process known as [MASK]."
inputs = tokenizer(prompt, return_tensors="pt").to("cuda")

with torch.no_grad():
    outputs = model(**inputs)
    mask_token_index = torch.where(inputs["input_ids"] == tokenizer.mask_token_id)[1]
    predictions = outputs.logits[0, mask_token_index]

top_k = torch.topk(predictions, k=5).indices.tolist()
for token_id in top_k[0]:
    print(f"Prediction: {tokenizer.decode([token_id])}")
```

**Review by stevhliu — COMMENTED** [review]
Nice, thanks for your contribution!

File: docs/source/en/model_doc/albert.md (hunk 1 above)
Comment (souvikchand) [reply docs/source/en/model_doc/albert.md:133]: Since almost every model's model card will have an `AttentionMaskVisualizer` but ALBERT is not getting it, isn't it good to explain why it don't have any?

File: docs/source/en/model_doc/albert.md (hunk 4 above)
Comment (souvikchand) [reply docs/source/en/model_doc/albert.md:42]: resolved

File: docs/source/en/model_doc/albert.md (hunk 1 above)
Comment (stevhliu) [reply docs/source/en/model_doc/albert.md:127]: The `AttentionMaskVisualizer` has only been implemented for ~50/300 models so far so I think its okay to leave it out at the moment.

Comment (souvikchand) [reply docs/source/en/model_doc/albert.md:127]: ok alright. thanks for your response.

File: docs/source/en/model_doc/albert.md (hunk 5 above)
Comment (souvikchand) [reply docs/source/en/model_doc/albert.md:58]: it is done

Comment (souvikchand) [reply docs/source/en/model_doc/albert.md:58]: instead of `.to("cuda")` i wrote `.to(model.device)` . I hope it's okay

File: docs/source/en/model_doc/albert.md (hunk 3 above)
Comment (souvikchand) [reply docs/source/en/model_doc/albert.md:102]: could you please briefly explain why it's better not to mention internal differences like **SOP** or `token_type_ids` in the model card? I want to make sure I follow the best style when contributing next time.

File: docs/source/en/model_doc/albert.md [hunk 6]
```diff
@@ -14,99 +14,92 @@ rendered properly in your Markdown viewer.
...
 ```
-from transformers import AlbertModel
-model = AlbertModel.from_pretrained("albert/albert-base-v1", torch_dtype=torch.float16, attn_implementation="sdpa")
-...
+
+</hfoption>
+<hfoption id="transformers-cli">
+
+```bash
+echo -e "Plants create [MASK] through a process known as photosynthesis." | transformers-cli run --task fill-mask --model albert-base-v2 --device 0
```
Comment (stevhliu) [inline docs/source/en/model_doc/albert.md:86]: Following #37657, lets also update the `transformers` command

```suggestion
<hfoption id="transformers CLI">

```bash
echo -e "Plants create [MASK] through a process known as photosynthesis." | transformers run --task fill-mask --model albert-base-v2 --device 0
```

File: docs/source/en/model_doc/albert.md (hunk 3 above)
Comment (stevhliu) [reply docs/source/en/model_doc/albert.md:102]: This is more useful background knowledge versus practical tips for a developer using the model so thats why we don't need to really mention the pretraining objective etc.

**Review by stevhliu — COMMENTED** [review]
Thanks, just a few more minor details :)

File: docs/source/en/model_doc/albert.md (hunk 3 above)
Comment (souvikchand) [reply docs/source/en/model_doc/albert.md:102]: ok. thanks for your guidance.

**Comment by souvikchand** [comment]
@stevhliu, I have done suggested changes. please review and let me know if it is ok 😄

**Comment by stevhliu** [comment]
Hey, sorry for the delay! PR looks good to me, feel free to mark it as ready for review whenever you're ready 🙂

**Comment by stevhliu** [comment]
Hi, would you mind rebasing your branch?

**Comment by souvikchand** [comment]
@stevhliu, I guess I made the rebase. Since I am new to this can you please confirm if I have made it? 
Also sorry for the delay 😓

**Review by stevhliu — APPROVED** [review]
Thanks and no problem, this looks good!

File: docs/source/en/model_doc/albert.md [hunk 7]
```diff
@@ -14,100 +14,100 @@ rendered properly in your Markdown viewer.
...
+
+<<<<<<< HEAD
+ALBERT was created to address problems like -- GPU/TPU memory limitations, longer training times, and unexpected model degradation in BERT. ALBERT uses two parameter-reduction techniques to lower memory consumption and increase the training speed of BERT:
+
+- **Factorized embedding parameterization:** The large vocabulary embedding matrix is decomposed into two smaller matrices, reducing memory consumption.
+- **Cross-layer parameter sharing:** Instead of learning separate parameters for each transformer layer, ALBERT shares parameters across layers, further reducing the number of learnable weights.
+
+ALBERT uses absolute position embeddings (like BERT) so padding is applied at right. Size of embeddings is 128 While BERT uses 768. ALBERT can processes maximum 512 token at a time. 
+>>>>>>> 7ba1110083 (Update docs/source/en/model_doc/albert.md
)
```
Comment (druvdub) [inline docs/source/en/model_doc/albert.md:40]: Hey @stevhliu, I found some merge conflict artifacts that weren't removed during the PR merge. These lines are still showing in the docs. Should I open a new PR to clean these up?

image

Comment (druvdub) [reply docs/source/en/model_doc/albert.md:40]: I am making a PR for the docs for auto classes, so I could push the fix with that?

Comment (stevhliu) [reply docs/source/en/model_doc/albert.md:40]: Oops, good catch! Can you just raise a separate PR to fix the merge conflict artifacts please?

File: docs/source/en/model_doc/albert.md [hunk 8]
```diff
@@ -14,100 +14,144 @@ rendered properly in your Markdown viewer.
...
-speed of BERT:
+ALBERT was created to address problems like -- GPU/TPU memory limitations, longer training times, and unexpected model degradation in BERT. ALBERT uses two parameter-reduction techniques to lower memory consumption and increase the training speed of BERT:
 
-- Splitting the embedding matrix into two smaller matrices.
-- Using repeating layers split among groups.
+- **Factorized embedding parameterization:** The large vocabulary embedding matrix is decomposed into two smaller matrices, reducing memory consumption.
+- **Cross-layer parameter sharing:** Instead of learning separate parameters for each transformer layer, ALBERT shares parameters across layers, further reducing the number of learnable weights.
 
-The abstract from the paper is the following:
+ALBERT uses absolute position embeddings (like BERT) so padding is applied at right. Size of embeddings is 128 While BERT uses 768. ALBERT can processes maximum 512 token at a time.
```
[the remark this answers is not in the document]
Comment (druvdub) [reply docs/source/en/model_doc/albert.md:39]: Hey @stevhliu, does this still need to go in `notes` section
